### PR TITLE
Add headphones-toolbox to nixpkgs

### PR DESCRIPTION
**Pull request by lambdachad**
## Description of changes

This PR adds the headphones-toolbox package, which is a UI for configuring Ploopy Headphones made with Tauri. Note that this doesn't add the required udev rules which must exist to make it work: https://github.com/george-norton/headphones-toolbox#quickstart-guide

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Move `pkgs/applications/audio/headphones-toolbox/default.nix` to `pkgs/by-name/he/headphones-toolbox/package.nix`, if you don't want to touch `pkgs/top-level/all-packages.nix`.

File: pkgs/applications/audio/headphones-toolbox/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+, stdenv
+, dpkg
+, fetchurl
+, autoPatchelfHook
+, wrapGAppsHook
+, webkitgtk
+, glib-networking
+, libappindicator
+, libayatana-appindicator
+, gst_all_1
```
Comment (wegank) [inline pkgs/applications/audio/headphones-toolbox/default.nix:11]:
```suggestion
, webkitgtk
```

File: pkgs/applications/audio/headphones-toolbox/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A UI for configuring Ploopy Headphones";
+    homepage = "https://github.com/george-norton/headphones-toolbox";
+    maintainers = with maintainers; [ knarkzel nyanbinary ];
+    license = licenses.gpl3Only;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    platforms = platforms.linux;
```
Comment (wegank) [inline pkgs/applications/audio/headphones-toolbox/default.nix:53]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

File: pkgs/applications/audio/headphones-toolbox/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  version = "0.0.3";
+
+  src = fetchurl {
+    url = "https://github.com/george-norton/headphones-toolbox/releases/download/headphones-toolbox-beta-v4r2/ploopy-headphones-toolbox_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-r+ybcD6koSIJ/6cck3RNXmf758sRnhS1Y4kaYCNbveA=";
+  };
+
+  unpackCmd = ''
+    dpkg-deb -x $curSrc source
+  '';
```
Comment (wegank) [inline pkgs/applications/audio/headphones-toolbox/default.nix:25]:
```suggestion
```

**Comment by knarkzel** [comment]
Resolved all of those and moved it.

File: pkgs/by-name/he/headphones-toolbox/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    mv usr/lib $out
+    mv usr/share $out
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A UI for configuring Ploopy Headphones";
+    homepage = "https://github.com/george-norton/headphones-toolbox";
+    maintainers = with maintainers; [ knarkzel nyanbinary ];
```
Comment (wegank) [inline pkgs/by-name/he/headphones-toolbox/package.nix:41]: Have you added yourself to the maintainer list?

Comment (Nyabinary) [reply pkgs/by-name/he/headphones-toolbox/package.nix:41]: @wegank Yeah and me as well :)

Comment (wegank) [reply pkgs/by-name/he/headphones-toolbox/package.nix:41]: I mean in `maintainers/maintainer-list.nix`. This PR clearly did not pass that check with a big red X.

Comment (knarkzel) [reply pkgs/by-name/he/headphones-toolbox/package.nix:41]: Added to maintainer list

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -18227,7 +18239,12 @@
     email = "elisp.vim@gmail.com";
     github = "vyp";
     githubId = 3889405;
-    name = "vyp";
+    name = "vyp";nyanbinary = {
+    email = "vextium@skiff.com";
+    github = "nyabinary";
+    githubId =     97130632;
+    name = "Niko";
+}
```
Comment (wegank) [inline maintainers/maintainer-list.nix:18247]:
```suggestion
    name = "vyp";
```

Already done on top.

**Comment by wegank** [comment]
Please squash the commits into two, in the following order:

1. `maintainers: add knarkzel and nyanbinary`
2. `headphones-toolbox: init at 0.0.3`

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -9199,6 +9199,12 @@
     github = "KnairdA";
     githubId = 498373;
   };
+  knarkzel = {
+    email = "knarkzel@gmail.com";
+    github = "zzzsyyy";
```
Comment (zzzsyyy) [inline maintainers/maintainer-list.nix:9204]: This seems like a mistake

Comment (knarkzel) [reply maintainers/maintainer-list.nix]: Yeah uhh

**Comment by knarkzel** [comment]
I made a mistake while squashing, so uhh I will close this.

**Comment by knarkzel** [comment]
Here's the latest one:

```nix
{ lib
, stdenv
, dpkg
, fetchurl
, autoPatchelfHook
, webkitgtk
}:

stdenv.mkDerivation (finalAttrs: {
  name = "headphones-toolbox";
  version = "0.0.3";

  src = fetchurl {
    url = "https://github.com/george-norton/headphones-toolbox/releases/download/headphones-toolbox-beta-v4r2/ploopy-headphones-toolbox_${finalAttrs.version}_amd64.deb";
    hash = "sha256-r+ybcD6koSIJ/6cck3RNXmf758sRnhS1Y4kaYCNbveA=";
  };

  unpackCmd = ''
    dpkg-deb -x $curSrc source
  '';

  nativeBuildInputs = [
    dpkg
    autoPatchelfHook
  ];

  buildInputs = [
    webkitgtk
  ];

  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
    mv usr/bin $out
    mv usr/lib $out
    mv usr/share $out

    runHook postInstall
  '';

  meta = with lib; {
    description = "A UI for configuring Ploopy Headphones";
    homepage = "https://github.com/george-norton/headphones-toolbox";
    maintainers = with maintainers; [ knarkzel nyanbinary ];
    license = licenses.gpl3Only;
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
    platforms = [ "x86_64-linux" ];
    mainProgram = "headphones-toolbox";
  };
})
```

**Comment by knarkzel** [comment]
Opened new one here: https://github.com/NixOS/nixpkgs/pull/254611

**Pull request closed without merging**